apps/dashboard/src/app/(dashboard)/dashboard/connect/ecosystem/[ecosystemId]/components/client/integration-permissions-section.client.tsx

```diff
@@ -0,0 +1,78 @@
+"use client";
+import { RadioGroup, RadioGroupItem } from "@/components/ui/radio-group";
+import { cn } from "@/lib/utils";
+import { useState } from "react";
+import { toast } from "sonner";
+import type { Ecosystem } from "../../../types";
+import { useUpdateEcosystem } from "../../hooks/use-update-ecosystem";
+
+export function IntegrationPermissionsSection({
+  ecosystem,
+}: { ecosystem: Ecosystem }) {
+  const [inflightValue, setInflightValue] = useState<
+    Ecosystem["permission"] | undefined
+  >();
+  const { updateEcosystem } = useUpdateEcosystem({
+    onMutate: (variables) => {
+      // When the mutation is called, set inflight value to the current value for optimistic UI update
+      setInflightValue(variables.permission);
+    },
```

Comment:
hmm, there's no way to get the inflight variables directly from the mutation? feels like there should be, this setState bothers me here